contents/docs/how-posthog-works/recordings-ingestion.mdx

```diff
@@ -0,0 +1,130 @@
+---
+title: Session recordings ingestion
+---
+
+## Overview
+
+We use `rrweb` to collect "snapshot data" from the browser. This data is gathered by the capture API and sent to ingestion.
+
+Because we use kafka for ingestion there is a maximum allowed message size and snapshot data is often larger than that. 
+
+So, in Capture we compress the data and split it into chunks. The plugin server runs a "recordings ingestion" workload which uses our ingestion pipeline to send each recording event for storage in ClickHouse
+
+```mermaid
+flowchart LR
+    clickHouse_session_recording_events_table[(Session recording events table)]
+    received_snapshot_event[clients send snapshot data]
+    kafka_recordings_topic>Kafka recordings topic]
+    kafka_clickhouse_recordings_topic>Kafka clickhouse recordings topic]
+
+    subgraph recordings ingestion workload
+        recordings_consumer-->kafka_clickhouse_recordings_topic
+        kafka_clickhouse_recordings_topic-->clickHouse_session_recording_events_table
+    end
+
+    received_snapshot_event --> capture
+    capture -->|compress and chunk to fit into kafka messages| kafka_recordings_topic
+    kafka_recordings_topic --> recordings_consumer
+```
+
+## Evolution to blob storage
+
+This is in progress.
+
+Storing snapshot data in ClickHouse is not ideal. It's not a good fit for the data model and it's not a good fit for the workload. So, recordings storage is being moved to blob storage. We use [minio](https://min.io/product/kubernetes) for blob storage. 
```

Comment:
No we don't 😅 We only use Minio for local dev and basically never recommend it over a cloud native S3 solution so I think we should say that.
```suggestion
Storing snapshot data in ClickHouse is not ideal. It's not a good fit for the data model and it's not a good fit for the workload. So, recordings storage is being moved to blob storage. We use AWS S3 for blob storage but in theory any S3 compatible API such as Minio would work as well. 
```

Reply:
#TIL